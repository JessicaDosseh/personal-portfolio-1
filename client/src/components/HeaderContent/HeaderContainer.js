import React from "react";
import styled, { keyframes, css } from "styled-components";
import ahLogo from "../../images/ah-logo.png";
import ahLogoWhite from "../../images/ah-logo-white.png";
import lambdaLogo from "../../images/lambda-logo.png";
import lambdaLogoWhite from "../../images/lambda-logo-white.png";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";

const hoverLink = keyframes`
  0% {
    color: #5A22DB;
  }
  100% {
    color: #AD91ED;
  }
`;

const hoverLinkDarkmode = keyframes`
  0% {
    color: #1a97f0;
  }
  100% {
    color: #6DBCF5;
  }
`;

const HeaderContainerDiv = styled.div`
  width: 100%;
  max-width: 1200px;
  padding: 50px 0;
  margin: 0 auto;

  ${props =>
    props.darkmode &&
    css`
      color: white;
    `}

  p {
    max-width: 1000px;
    width: 100%;

    a {
      font-size: 3rem;
      color: #6936de;

      ${props =>
        props.darkmode &&
        css`
          color: #1a97f0;
        `}

      &:hover {
        ${props =>
          props.darkmode
            ? css`
                animation: ${hoverLinkDarkmode} 0.5s forwards;
              `
            : css`
                animation: ${hoverLink} 0.5s forwards;
              `}
      }

      &:active {
        ${props =>
          props.darkmode
            ? css`
                color: #b6465f;
              `
            : css`
                color: #ad91ed;
              `}
      }
    }
  }
`;

const CurrentPreviousWork = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;

  @media (min-width: 800px) {
    flex-direction: row;
  }

  div {
    h4 {
      letter-spacing: 4px;
      margin-top: 50px;

      @media (min-width: 800px) {
        margin-top: 100px;
      }
    }

    div {
      padding: 10px 0;
      width: 60%;
      max-width: 300px;
      opacity: 1;
      transition: 0.2s;
      margin: 0;

      @media (min-width: 800px) {
        width: 300px;
      }

      &:hover {
        opacity: 0.8;
        padding: 6px 0 14px;
      }
    }
  }

  div:first-child {
    @media (min-width: 500px) {
      padding-right: 5%;
    }

    @media (min-width: 800px) {
      padding-right: 6.5%;
    }
  }
`;

const HeaderContainer = props => {
  return (
    <HeaderContainerDiv darkmode={props.darkmode}>
      <p>
        I'm Nathan Thomas, a{" "}
        <a href="https://github.com/nwthomas">full stack software engineer</a>{" "}
        living in the Napa Valley and currently attending{" "}
        <a href="https://lambdaschool.com/">Lambda School</a>. Let's{" "}
        <Link to="/contact">talk</Link>.
      </p>
      <CurrentPreviousWork>
        <div>
          <h4>Current Student:</h4>
          <div>
            <a href="https://lambdaschool.com/">
              <img
                data-testid="schoolLogo"
                src={props.darkmode ? lambdaLogoWhite : lambdaLogo}
                alt="Current school logo"
              />
            </a>
          </div>
        </div>
        <div>
          <h4>Previously:</h4>
          <div>
            <a href="https://www.adventisthealth.org/">
              <img
                data-testid="workLogo"
                src={props.darkmode ? ahLogoWhite : ahLogo}
                alt="Former employer logo"
              />
            </a>
          </div>
        </div>
      </CurrentPreviousWork>
    </HeaderContainerDiv>
  );
};

HeaderContainer.propTypes = {
  darkmode: PropTypes.bool.isRequired
};

export default HeaderContainer;
