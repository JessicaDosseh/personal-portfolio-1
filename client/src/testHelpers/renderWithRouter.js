import React from "react";
import { Router } from "react-router-dom";
import { render } from "react-testing-library";
import { createMemoryHistory } from "history";
import { StateProvider } from "react-conflux";
import { darkContext } from "../store/contexts";
import { globalReducer } from "../store/reducers/globalReducer";

const renderWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...render(
      <StateProvider reducer={globalReducer} stateContext={darkContext}>
        <Router history={history}>{ui}</Router>
      </StateProvider>
    ),
    history
  };
};

export default renderWithRouter;
