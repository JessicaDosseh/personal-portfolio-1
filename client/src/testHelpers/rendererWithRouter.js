import React from "react";
import { Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { createMemoryHistory } from "history";
import { StateProvider } from "react-conflux";
import { darkContext } from "../store/contexts";
import { globalReducer } from "../store/reducers/globalReducer";

const rendererWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...renderer.create(
      <StateProvider reducer={globalReducer} stateContext={darkContext}>
        <Router history={history}>{ui}</Router>
      </StateProvider>
    ),
    history
  };
};

export default rendererWithRouter;
