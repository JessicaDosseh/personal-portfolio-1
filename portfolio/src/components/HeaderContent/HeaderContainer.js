import React from "react";
import styled, { keyframes } from "styled-components";
import ahLogo from "../../images/ah-logo.png";
import lambdaLogo from "../../images/lambda-logo.png";

const hoverLink = keyframes`
  0% {
    color: ##5A22DB;
  }
  100% {
    color: #AD91ED;
  }
`;

const hoverPhoto = keyframes`
  0% {
    opacity: 0.65;
  }
  100% {
    opacity: 1;
  }
`;

const HeaderContainerDiv = styled.div`
  width: 90%;
  max-width: 1100px;
  margin: 50px auto;

  p {
    font-size: 3rem;
    line-height: 1.8;

    a {
      font-size: 3rem;
      color: #6936de;

      &:hover {
        animation: ${hoverLink} 0.5s forwards;
      }
    }
  }

  div {
    h4 {
      letter-spacing: 4px;
      margin-top: 50px;
    }
    div {
      margin: 10px 0;
      width: 60%;
      opacity: 0.65;

      &:hover {
        animation: ${hoverPhoto} 0.5s forwards;
      }
    }
  }
`;

const HeaderContainer = () => {
  return (
    <HeaderContainerDiv>
      <p>
        I'm Nathan Thomas, a{" "}
        <a href="https://github.com/nwthomas">full stack software engineer</a>{" "}
        living in the Napa Valley and currently enrolled at{" "}
        <a href="https://lambdaschool.com/">Lambda School</a>.
      </p>
      <div>
        <h4>CURRENT STUDENT:</h4>
        <div>
          <a href="https://lambdaschool.com/">
            <img src={lambdaLogo} alt="Current school logo" />
          </a>
        </div>
        <h4>PREVIOUSLY:</h4>
        <div>
          <a href="https://www.adventisthealth.org/">
            <img src={ahLogo} alt="Former employer logo" />
          </a>
        </div>
      </div>
    </HeaderContainerDiv>
  );
};

export default HeaderContainer;
