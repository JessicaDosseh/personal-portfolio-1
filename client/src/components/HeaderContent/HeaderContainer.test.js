import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { renderWithRouter, rendererWithRouter } from "../../testHelpers";

import HeaderContainer from "./HeaderContainer";

describe("<HeaderContainer />", () => {
  it("should render the HeaderContainer component without crashing", () => {
    const helpers = renderWithRouter(<HeaderContainer darkmode={false} />);
  });

  it("should match the snapshot of HeaderContainer", () => {
    const tree = rendererWithRouter(<HeaderContainer darkmode={true} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("should display the header content text", () => {
    const { getByText } = renderWithRouter(<HeaderContainer darkmode={true} />);
    expect(getByText(/I'm Nathan Thomas/i)).toBeDefined();
  });

  it("should render the logos", () => {
    const { getByTestId } = renderWithRouter(
      <HeaderContainer darkmode={true} />
    );
    expect(getByTestId("schoolLogo")).toBeDefined();
    expect(getByTestId("twitterLogo")).toBeDefined();
  });
});
