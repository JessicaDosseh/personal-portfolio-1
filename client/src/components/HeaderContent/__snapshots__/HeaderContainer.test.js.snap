// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderContainer /> should match the snapshot of HeaderContainer 1`] = `
<div
  className="sc-bdVaJa jKTdSj"
>
  <p>
    I'm Nathan Thomas, a
     
    <a
      href="https://github.com/nwthomas"
    >
      software engineer
    </a>
     living in San Francisco and working at 
    <a
      href="https://twitter.com/"
    >
      Twitter
    </a>
    . Let's 
    <a
      href="/contact"
      onClick={[Function]}
    >
      talk
    </a>
    .
  </p>
  <div
    className="sc-bwzfXH kLAFPo"
  >
    <div>
      <h4>
        Currently:
      </h4>
      <div>
        <a
          className="twitter-logo"
          href="https://twitter.com/"
        >
          <img
            alt="Twitter logo"
            data-testid="twitterLogo"
            src="twitter-logo-white.png"
          />
        </a>
      </div>
    </div>
    <div>
      <h4>
        Previously:
      </h4>
      <div>
        <a
          href="https://lambdaschool.com/"
        >
          <img
            alt="Current school logo"
            data-testid="schoolLogo"
            src="lambda-logo-white.png"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
